geo/src/types.rs

```diff
@@ -37,6 +37,16 @@ impl<F: GeoFloat> Closest<F> {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum GeoError {
```

Comment:
> I'd guess we want this to be non-exhaustive

I mean like this:

```suggestion
#[non_exhaustive]
pub enum GeoError {
```

That way, if we need to add new error cases in a subsequent release, we can do so without breaking existing user's code.